src/Client.js

```diff
@@ -134,12 +135,37 @@ class Client extends EventEmitter {
     /**
      * Send a message to a specific chatId
      * @param {string} chatId
-     * @param {string} message 
+     * @param {string} message
      */
     async sendMessage(chatId, message) {
-        await this.pupPage.evaluate((chatId, message) => {
-            Store.SendMessage(Store.Chat.get(chatId), message);
-        }, chatId, message)
+        let msg = {};
+        if (message && typeof message === 'object' && message.constructor === Object) {
+            msg = message;
+        } else {
+            msg = {
+                linkPreview: null,
+                mentions: [],
+                quotedMsg: null,
+                quotedMsgAdminGroupJid: null,
+                body: message
+            }
+        }
+        let last_message = await this.pupPage.evaluate(async (chatId, msg) => {
+            msg.mentionedJidList = [];
+            for (let i = 0; i < msg.mentions.length; i++) {
+                let user = await Store.Contact.serialize().find(x => x.id.user === msg.mentions[i].split("@")[0]);
```

Comment:
Since you're using Contact, would this work if the user isn't one of your contacts / if you don't have a chat open with them?

Also, maybe we should just pass an extra argument to the sendMessage function that's an object with these other properties, leaving the first argument as a string as is